tests/python/pants_test/engine/scheduler_test_base.py

```diff
@@ -97,3 +100,120 @@ def execute_raising_throw(self, scheduler, product, subject):
     self.assertTrue(type(resulting_value) is Throw)
 
     raise resulting_value.exc
+
+
+class DeclarativeTaskTestBase(SchedulerTestBase):
```

Comment:
It should at this point be possible to extend `TestBase` (and or `TaskTestBase`) to get a fully useful, legacy-configured `Scheduler` via `self.scheduler`... `SchedulerTestBase` should be primarily useful for tests that intend to manually construct their own scheduler.

Reply:
Removed all the scheduler stuff and moved to `task_test_base.py`!